example/build_cmake_installed_kk_as_language/CMakeLists.txt

```diff
@@ -8,10 +8,11 @@ project(Example CXX Fortran CUDA)
 
```

Comment:
@dalg24 rightfully pointed out that we can make it fully backend-agnostic by cmake's `enable_language`. Will be added

Reply:
done in https://github.com/kokkos/kokkos/pull/7525/commits/5c2d3b84b52992679a1c36cd2c750aa16b647cd5

Reply:
this actually does not work as `enable_language` can not be called inside a function. We could make `kokkos_set_source_language` a macro, but I am hesitant because of the implications.

Thus the PR proposes the `kokkos_enable_language` macro that does nothing except enabling the language

Reply:
I think it's fine for users to call
```
enable_language(${Kokkos_COMPILE_LANGUAGE})
```
instead of `kokkos_enable_language()`. I could argue either way.

Reply:
afaik, we don't export `Kokkos_COMPILE_LANGUAGE` the user would need some logic to extract it from `Kokkos_DEVICES`.

Reply:
We do https://github.com/kokkos/kokkos/blob/7ff4042fd2dc1d1fe6ae280179fc693ad6dac6ac/cmake/KokkosConfig.cmake.in#L66.

Reply:
how did I miss that? I ditched the macro and have the user call `enable_language`

Reply:
We could have both